src/main/java/com/epam/webapphello/command/LoginCommand.java

```diff
@@ -0,0 +1,43 @@
+package com.epam.webapphello.command;
+
+import com.epam.webapphello.entity.User;
+import com.epam.webapphello.exception.CommandException;
+import com.epam.webapphello.exception.ServiceException;
+import com.epam.webapphello.service.UserService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
+
+public class LoginCommand implements Command {
+
+    private final UserService userService;
+
+    public LoginCommand(UserService userService) {
+        this.userService = userService;
+    }
+
+
+    @Override
+    public String execute(HttpServletRequest req, HttpServletResponse resp) throws CommandException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+
+        Optional<User> user = null;
+
+        try {
+            user = userService.login(login, password);
+        } catch (ServiceException e) {
+            throw new CommandException(e);
```

Comment:
no need in command exception, just thow original service exception

Reply:
i think that block Command and block Service are different as Dao and Service, because block Command connect with jsp and servlet, block Service - mediator between Servlet(Command) and Dao. So i decided that Command is not part of Service and crete CommandException

Reply:
Ok